Microsoft.Alm.Authentication/Secret.cs

```diff
@@ -54,6 +54,22 @@ public static string UriToUrl(TargetUri targetUri, string @namespace)
             return targetName;
         }
 
+        /// <summary>
+        ///     Generate a key based on the ActualUri.
+        ///     This may include username, port, etc
+        /// </summary>
+        public static string UriToActualUri(TargetUri targetUri, string @namespace)
+        {
+            BaseSecureStore.ValidateTargetUri(targetUri);
+            if (String.IsNullOrWhiteSpace(@namespace))
+                throw new ArgumentNullException(@namespace);
+
+            string targetName = $"{@namespace}:{targetUri.ActualUri.AbsoluteUri}";
```

Comment:
Why use `.ActualUri` here? Why not `targetUri.ToString(false, false, false)` or similar?

Seems like `.ActualUri` should be fine, but I'd like to understand the decision.